### PR TITLE
Moves links to wayback machine, fixes one bad `s/redis/valkey/`

### LINKED ISSUE #188: Antirez blog forwards to redis, lacks valid cert, switch to wayback machine links
The docs reference Salvatore's blog liberally for both historical context and technical information.

It's been a long term, known issue that the blog lacks a valid certificate and has a scary "Not Secure" warning and is only accessible via `http`, not `https`. 

I noticed recently that my browser (Chrome) tried to do a move to `https` when manually pasting an `http` antirez.com URL into the address bar; https forwards to the redis.io domain which resolves to the redis corporate page (and results in a broken link). It seems that Chrome will, for now, follow http _links_ in a page.

Indeed, `https` links on antirez.com are being 301 redirected to redis.io, as seen by `wget`:

```
$ wget https://antirez.com/news/133 --no-check-certificate
--2024-11-20 07:40:10--  https://antirez.com/news/133
Resolving antirez.com (antirez.com)... 178.62.37.191
Connecting to antirez.com (antirez.com)|178.62.37.191|:443... connected.
WARNING: cannot verify antirez.com's certificate, issued by ‘CN=Let's Encrypt Authority X3,O=Let's Encrypt,C=US’:
  Unable to locally verify the issuer's authority.
WARNING: no certificate subject alternative name matches
	requested host name ‘antirez.com’.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://redis.io/news/133 [following]
--2024-11-20 07:40:11--  https://redis.io/news/133
Resolving redis.io (redis.io)... 45.60.123.1, 45.60.129.1
Connecting to redis.io (redis.io)|45.60.123.1|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-11-20 07:40:12 ERROR 404: Not Found.
```

I think it might be wise to move antirez.com links to wayback machine links.

E.g. http://antirez.com/news/133 -> https://web.archive.org/web/20241019222138/http://antirez.com/news/133

This resolves the current issue of `http` warnings in browsers and any potential forwarding shenanigans in the future.

### PR DESCRIPTION
**Pull request by stockholmux**
This PR moves links to antriez.com to the most recent wayback machine links. This ensures that the content that the docs point to is the intended content, it has proper https, and doesn't unexpectedly forward.

Additionally, there was one improper search/replace that was reworded when the link was changed. 

Fixes #188

### CODE REVIEW

File: topics/distlock.md [hunk 1]
```diff
@@ -228,4 +228,4 @@ Thanks in advance!
 ## Analysis of Redlock
 ---
 
-1. Martin Kleppmann [analyzed Redlock here](https://martin.kleppmann.com/2016/02/08/how-to-do-distributed-locking.html). A counterpoint to this analysis can be [found here](http://antirez.com/news/101).
+1. Martin Kleppmann [analyzed Redlock here](https://martin.kleppmann.com/2016/02/08/how-to-do-distributed-locking.html). A counterpoint to this analysis can be [found here](https://web.archive.org/web/20241209045142/https://antirez.com/news/101).
```
Comment (zuiderkwast) [inline topics/distlock.md:231]: https in this one? It's http in all the others

**Review by zuiderkwast — COMMENTED** [review]
LGTM, just some nits about minor differences in the links.

File: topics/admin.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ description: Advice for configuring and managing Valkey in production
 
 ### Security
 
-* By default, Valkey does not require any authentication and listens to all the network interfaces. This is a big security issue if you leave Valkey exposed on the internet or other places where attackers can reach it. See for example [this attack](http://antirez.com/news/96) to see how dangerous it can be. Please check our [security page](security.md) and the [quick start](quickstart.md) for information about how to secure Valkey.
+* By default, Valkey does not require any authentication and listens to all the network interfaces. This is a big security issue if you leave Valkey exposed on the internet or other places where attackers can reach it. See for example [this attack](https://web.archive.org/web/20241019230117/http://www.antirez.com/news/96) to see how dangerous it can be. Please check our [security page](security.md) and the [quick start](quickstart.md) for information about how to secure Valkey.
```
[the remark this answers is not in the document]
Comment (stockholmux) [reply topics/admin.md:41]: huh. Weird. I just copy and pasted these. No idea how the www got in there .

File: topics/distlock.md (hunk 1 above)
Comment (stockholmux) [reply topics/distlock.md:231]: Weird.

I think what happened is that I just pasted the links into wayback machine and use the right arrow to get the latest. I guess this got the https version.

I'm glad to change it but I don't think it has any bearing on what's being served.

**Review by zuiderkwast — APPROVED** [review]

**Pull request merged**